### ISSUE TITLE
pub build: javascript lacks release information and sourceMappingURL

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub#765]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [kevmoo](https://github.com/kevmoo)**
_Originally opened as dart-lang/sdk#15563_

----

Using Pub version 1.1.0-edge.31027

But with the attached installed dependencies

I'm getting the map files generated, which is great, but I need to manually add sourceMappingURL at the bottom of the generated JS to get Chrome to see them

...also nice to have the compiler version at the top, but that's less important
______
**Attachment:**
[pubspec.lock](https://storage.googleapis.com/google-code-attachments/dart/issue-15563/comment-0/pubspec.lock) (2.37 KB)

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

Bouncing over to dart2js since this is a request to add something to the generated JS.
______
*Removed [Library-Barback](../labels/Library-Barback) label.*
*Added [Area-Dart2JS](../labels/Area-Dart2JS) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kevmoo](https://github.com/kevmoo)**

----

FYI: when I do a normal dart2js (even minimized) I still get the build info at the top and sourceMappingURL at the bottom.

Curious why the output from pub/barback would lack this.

Is it something w/ options passed (or not passed) into dart2js?

**Comment by DartBot** [comment]
<a href="https://github.com/peter-ahe-google"><img src="https://avatars.githubusercontent.com/u/5689005?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [peter-ahe-google](https://github.com/peter-ahe-google)**

----

*Removed [Area-Dart2JS](../labels/Area-Dart2JS) label.*
*Added [Area-Pub](../labels/Area-Pub) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/peter-ahe-google"><img src="https://avatars.githubusercontent.com/u/5689005?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [peter-ahe-google](https://github.com/peter-ahe-google)**

----

The user of the compiler library decides were to put the source maps. Consequently, the compiler library expects that the user appends the sourceMappingURL.

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

> Consequently, the compiler library expects that the user appends the sourceMappingURL.

Ah, OK. We can handle that then. Thanks for clarifying!

**Comment by DartBot** [comment]
<a href="https://github.com/peter-ahe-google"><img src="https://avatars.githubusercontent.com/u/5689005?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [peter-ahe-google](https://github.com/peter-ahe-google)**

----

I should probably mention that this is the current behavior.

Right now, the compiler API uses a "CompilerOutputProvider" to return an EventSink&lt;String>. It would make sense to change the API so that the compiler library also becomes aware of the URI corresponding to this EventSink so that it can information such as sourceMappingURL.

**Comment by DartBot** [comment]
<a href="https://github.com/peter-ahe-google"><img src="https://avatars.githubusercontent.com/u/5689005?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [peter-ahe-google](https://github.com/peter-ahe-google)**

----

Filed issue dart-lang/sdk#15607.

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

*Marked this as being blocked by dart-lang/sdk#15607.*

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

*Removed [Priority-Unassigned](../labels/Priority-Unassigned) label.*
*Added [Priority-Medium](../labels/Priority-Medium) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

*Marked this as blocking dart-lang/sdk#15829.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

Issue dart-lang/sdk#16472 has been merged into this issue.

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

----

Can I ask us to consider this issue for Milestone 1.2? I'm trying to integrate pub serve into my workflow, but this bug makes it very hard to do so. Without source maps, debugging is very difficult. Thanks for the consideration!

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

----

Also, trying to get a good mobile dev workflow, and pub serve could be a real win. But I extra-need source maps for mobile since we don't produce builds of Dartium for Android. :)

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

Seth: You might want to make this plea on issue dart-lang/sdk#15607, since that's where work needs to happen to get this supported.

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

----

Roger that, thanks for the pointer.

**Comment by DartBot** [comment]
<a href="https://github.com/peter-ahe-google"><img src="https://avatars.githubusercontent.com/u/5689005?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [peter-ahe-google](https://github.com/peter-ahe-google)**

----

*Marked this as being blocked by dart-lang/sdk#17259.*
*Unmarked this as being blocked by dart-lang/sdk#15607.*

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

*Set owner to @munificent.*
*Added [Started](../labels/Started) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

This is fixed now.

It should work with both pub serve and pub build. Note that with pub build, you'll need to do --mode debug to get our Dart source files copied over to the build output where the browser can find them.
______
*Added [Fixed](../labels/Fixed) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

----

Thanks Bob!
